snap/snapcraft.yaml

```diff
@@ -90,6 +90,15 @@ parts:
     stage:
     - -linux
     - -src
+    override-build: |
+      snapcraftctl build
+      if [ -d "../install/linux/lib64" ]; then
+        cd ../install/linux/lib64
+      else 
+        cd ../install/linux/lib32
+      fi
+      SO_FILE=$(find * -name "libphobos2.so.[0-9]*.[0-9]*.[0-9]*")
+      ln -sf $SO_FILE "${SO_FILE%.*}"
```

Comment:
Question: why isn't this symlink being created properly by `make install`?  Or is it being created properly but not preserved in the snap package?

I don't object to taking this change as a workaround but I'd like to confirm if we're not doing something hacky when the problem might actually be in how we're managing the install stage from the makefile.

Reply:
On my install I see:

```
libphobos2.a
libphobos2.so -> libphobos2.so.0.85.0 (symlink)
libphobos2.so.0.85.0
```

... which matches what is in the deb package available for download from dlang.org.  So what exactly is missing and why is it relevant/needed if upstream does not think it worth creating?

Reply:
These files are present in the deb package on dlang.org:
`libphobos2.a
libphobos2.so -> libphobos2.so.0.85.0
libphobos2.so.0.85 -> libphobos2.so.0.85.0
libphobos2.so.0.85.0`

We need libphobos2.so.0.85 because it is the soname of the shared library, I have no idea why make install does not create this symbolic link but Archlinux also uses another method to get pretty much the same thing (https://git.archlinux.org/svntogit/community.git/tree/trunk/PKGBUILD?h=packages/dmd#n126)

With this patch we would get this:
`libphobos2.a
libphobos2.so -> libphobos2.so.0.85.0
libphobos2.so.0.85.0`

to

`libphobos2.a
libphobos2.so -> libphobos2.so.0.85.0
libphobos2.so.0.85 -> libphobos2.so.0.85.0
libphobos2.so.0.85.0`

Reply:
dlang.org also seems to do the same thing (https://github.com/dlang/installer/blob/482840bf5 printe94ebb12aabaa7ddb0dac319be90/linux/dmd_phobos.sh#L169)